test/unit/util/detect.test.js

```diff
@@ -0,0 +1,59 @@
+import { afterEach, beforeEach, describe, it } from 'node:test';
+import { strict as assert } from 'node:assert';
+import * as Rapid from '../../../modules/headless.js';
+
+if (!global.window) {  // mock window for Node
```

Comment:
This seems ok, I've had to do some of these mocks in the rapid-sdk project too:
https://github.com/rapideditor/rapid-sdk/blob/c7618a4fa5e02e5d1421178f19828946830a746e/packages/util/src/session_mutex.ts#L8-L11

`global` will work ok here, but I think `globalThis` is the new cool way to do it, it works across more environments:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis